Sources/StreamChatUI/CommonViews/BaseViews.swift

```diff
@@ -6,6 +6,46 @@ import Foundation
 import StreamChat
 import UIKit
 
+// Protocol that provides accessibility features
+protocol AccessibilityView {
+    // Identifier for view
+    var accessibilityViewIdentifier: String { get }
+
+    // This function is called once the view is being added to the view hiearchy
+    func setAccessibilityIdentifier()
+}
+
+extension AccessibilityView where Self: UIView {
+    var accessibilityViewIdentifier: String {
+        "\(type(of: self))"
+    }
+
+    func setAccessibilityIdentifier() {
+        accessibilityIdentifier = accessibilityViewIdentifier
+    }
+}
+
+@propertyWrapper
+public struct AccessibleView<T: UIView> {
```

Comment:
All UI components instantiate their subviews **lazily** -> a property wrapper is not applicable for UI components except message content view but increases public API surface with smth not directly related to chat. Using internal `withAccessibilityIdentifier` in message content view feels more reasonable, tbh

Reply:
Hi @evsaev thanks for the commentt.

 All 3 cases how we're setting the accesibility identifiers are explained in the PR description. 


"All UI components instantiate their subviews lazily" - this claim is not true. There are cases when the views are not instantiated lazily, but rather kept `nil`. In those cases the property wrapper makes sense. It also allows us to leverage similar semantics in our custom apps. I see the benefit of this solution.

Reply:
It's only `ChatMessageCell/ChatMessageContentView` which are special, that's true for the rest of the UI codebase. Having multiple ways of doing the same thing (setting accessibility identifier in our case) is what we'd like to avoid as it's not obvious which way to choose 🙂

> similar semantics in our custom apps

Which apps do you mean?

Reply:
@evsaev Thanks for comment. There will always be at least 2-3 ways how & where to assign the accessibility identifier irregardless what helpers we provide.

The "chain of command" comes like this.
1. If the `View` class was declared in our SDK, it will share the identifier == to its class name
2. If the `View` is UIKit based and lazily instantiated -> we leverage the `with` helper
3. If the general `View` (irregardless of where it was defined) is declared we will leverage the property wrapper.

The rule of thumb is laziness of the engineer here. The accessibility identifiers should live as close to the property declaration as possible, so at glance you'd be able to say what's the accessibility identifier of given element. This comes with the time benefit when eg writing the UI tests for your new feature.

Reply:
@evsaev  I parked the `@propertyWrapper` for now.
The plan here is to leverage the testing suite also by our integrators. Giving us & them simple semantic to set accessibility identifier is valuable and help to spread the consistency and to keep the identifiers as close to the view declaration as possible.

Reply:
Meaning. We will provide that `propertyWrapper` at later stage. For now, we need to progress with this work. So guess reverting this functionality unblocks this PR.